src/main/java/org/broadinstitute/hellbender/utils/pairhmm/SampleSitesForDRAGstrModel.java

```diff
@@ -0,0 +1,437 @@
+                final int[] row = decimationMatrix[i];
+                result[i] = new long[row.length];
+                for (int j = 0; j < row.length; j++) {
+                    result[i][j] = (1 << row[j]) - 1;
+                }
+            }
+            return result;
+        }
+
+        public long mask(final int period, final int repeats) {
+            final int p = period >= decimationMask.length ? decimationMask.length - 1 : period;
+            final long[] masks = decimationMask[p];
+            if (masks.length == 0) {
+                return 0;
+            } else if (repeats >= masks.length) {
+                return masks[masks.length - 1];
+            } else {
+                return masks[repeats];
+            }
+        }
+
+        public boolean decimate(final int seqNumber, final int bestPeriod, final long bestPeriodRepeats) {
+            if (counts.length <= bestPeriod) {
+                return false;
+            } else {
+                final long[] periodCounts = counts[bestPeriod];
+                if (periodCounts.length == 0) {
+                    return false;
+                } else {
+                    final int effectiveRepeatCount
+                            = (int) (bestPeriodRepeats < periodCounts.length ? bestPeriodRepeats : periodCounts.length - 1);
+                    final long count = periodCounts[effectiveRepeatCount]++;
+                    final long left = count + seqNumber;
+                    final long right = decimationMask[bestPeriod][effectiveRepeatCount];
+                    return ((int) left & (int) right) != 0 || ((left >> 32) & (right >> 32)) != 0;
+                }
+            }
+        }
+    }
+
+    @Argument(fullName="decimation", doc="decimation per period and repeat. It can be \"DEFAULT\" to use the default values (default), " +
+            " \"NONE\" to deactivate decimation (potentially resulting in a very large output file) or indicate the path to a file" +
+            " that contains the decimation matrix.", optional = true)
+    private DecimationTable decimationTable = DecimationTable.DEFAULT;
+
+    @Argument(doc = "name of the zip file where the sites sampled will be stored",
+              fullName= StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+              shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME)
+    private String outputPath = null;
+
+    @Argument(fullName="max-period", doc="maximum STR period sampled", optional = true, minValue = 1, maxValue = 10)
+    private int maxPeriod = 8;
+
+    @Argument(fullName="max-repeats", doc="maximum STR repeat sampled", optional = true, minValue = 1, maxValue = 20)
+    private int maxRepeat = 20;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    private File tempDir;
+
+    private long[][] totalCounts;
+
+    private long[][] emittedCounts;
+
+    public void onStartup() {
+        super.onStartup();
+        try {
```

Comment:
Almost all of the code in this codepath is duplicated with SingleTextToDragstrLociZipBinary (which also doesn't have docs). You should only include one of these two classes and add a toggle to the main one in order to enable the binary intermediate files.